### PR TITLE
Replace icon svg with ones from svgrepo.com

### PR DESCRIPTION
**Pull request by Tampa**
This is more of a proposal to actually remove some of the flat, single color, icons that are very hard to see and offer very little recognition value. From a usability standpoint icons should have enough color and distinct shapes to make them recognizable.

While it is true that they might be problematic from an accessibility standpoint, that is generally problem the moment color is added to anything. Ideally for that a complete rewrite of the color management would be undertaken along with adding the ability to specify icon sets or even individual icons, but that's beyond the scope of this. Aim here is just to get the flat icons out since they are awful.

Icons taken from:

https://www.svgrepo.com/svg/276788/fingerprint
https://www.svgrepo.com/svg/424242/copybook-education-notebook https://www.svgrepo.com/svg/267738/puzzle
https://www.svgrepo.com/collection/small-flat-vectors/ https://www.svgrepo.com/collection/shipping-and-delivery-8/ https://www.svgrepo.com/svg/254993/stats
https://www.svgrepo.com/collection/soft-colored-ui-icons/ https://www.svgrepo.com/collection/flat-colored-interface-icons/ https://www.svgrepo.com/collection/computer-data/
https://www.svgrepo.com/collection/flat-icon-design-light-vectors/

All icons used under cc0 or public domain license.

Tried to stick to using just a few collections to create some sort of commonality between icons. Some were just not in there, so had to find single icons for some stuff.

None are final, if there are some that don't work then there are plenty of options to change. I picked the ones that I deemed most fit the original icons or the file name/description.

I would also like to fix the problem of "Edit View" and "Manage Jenkins" using the same cogwheel icon, but I don't know where that is defined. If someone could tell me that would be nice so I can amend this.

### Testing done

Manually replaced the icons in the computed html output. I have no build environment for java set up, but considering these are just svg there is little potential for issues. At most contrast issues depending on the icon. I use custom css to create a dark mode, svgrepo is on white background. On both the icons look readable enough though. If someone can build and spot me a binary to test I'd be happy to.

### Proposed changelog entries

- Changed icons for greater readability and usability

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@jenkinsci/sig-ux

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7786"></a>

### CODE REVIEW

File: war/src/main/resources/images/symbols/arrow-down.svg
Comment (daniel-beck) [inline war/src/main/resources/images/symbols/arrow-down.svg:1]: Did you add this one just to have a complete set?

**Comment by daniel-beck** [comment]
Testing this PR:

<details><summary>After setup</summary>

1-start

</details>

<details><summary>With a job</summary>

2-with-job

</details>

<details><summary>Viewing the job</summary>

3-job

</details>

<details><summary>Configuring the job</summary>

4-job-config

</details>

<details><summary>Manage Jenkins</summary>

5-manage

</details>

Personally, while I've been quite vocal about some of the drawbacks of design decisions made recently, this is a clear step backward for me, even ignoring that some icons are still in the current style. The icons simply look cheap. They're not _minimal_, but there are no gradients or other design elements beyond the minimum. The apparent attempt at a 3D look to the puzzle (plugin) icon looks just weird, especially at the smaller size used on Manage Jenkins.

File: war/src/main/resources/images/symbols/arrow-down.svg
Comment (Tampa) [reply war/src/main/resources/images/symbols/arrow-down.svg:1]: Pretty much. Bit odd that all other directions are in there but not downwards. For future use if it ever comes to that I guess.

**Comment by Tampa** [comment]
The icons are from multiple sets so they don't always fit together all that well. The plugin icon was one of the only ones I could find that really had more design to it. Mostly just puzzle pieces with very plain design.

Minimal design is horrible for any sort of usability. If all the icons look the same plain color on background you are required to read the text always. There just isn't enough distinct about them to be eye-catching enough otherwise. Some icons also, to me, don't make a whole lot of sense in what they are associated with. There are certainly more complex svg out there for these things, but at their scale these features are so small they might just all look the same anyways.

Like I said, only just a start. Some of the icons are re-used in various places so making a decision on what to use is made more difficult. In terms of their design language I went with ones that weren't overly complex since there seems to be this idea to go for simpler stuff. The ones still in the old style I am still not sure what to replace with, whether direct replacement or selecting something more representative of the context they are used in. Like Build triggers for example, drawing a blank on what shape I would expect there.

I struck up this topic on the gitter a couple times, but I never really got a good answer on what the whole idea is with regards to iconography. I tried to stick to as few sets as possible as to not mix too many different styles and keep things somewhat simple, adding a bit more shape and color, but not going overboard to stay away from the "cartoonish" style of the old icons. The newer UI doesn't really fit with those. I included the links in case someone else has an idea on what to use for something not yet changed. svgrepo seems to have a lot of icons of different styles that are under permissive license so there is plenty to select from. If I knew beforehand what style was wanted then would have gone for that.

Seems some of the svg lack data and don't display properly either. I did adjust two of them from their originals to bring them more in line with what they replaced.

Like some have mentioned ideally instead of shipping fixed icons there would be sets to select from or even just putting them directly into a css class so that changing them is as easy as adding custom css overrides.

It would help to have a concise design language and some information on what the whole plan is currently. I only know that these changes to the UI, adding lots of whitespace padding, flat colorless icons and button effects add nothing to readability, usability or overall look of the UI. I mentioned on gitter how that really reduced the usability for the developers I work with and where I could I turned their feedback into custom css, with the result of the UI of my instance currently looking like this: https://i.imgur.com/80jeHoQ.png Not pretty, but the overall return I got from that was "Yeah that works I guess", which tells me it is better than before, but not by much. Only so much css can do though, can't change the icons.

I got good feedback on gitter for proposed changes such as: https://i.imgur.com/7hrpSJ8.png Hence starting this project to actually replace all the icons. If the plan is another then feel free to close this request.

**Comment by janfaracik** [comment]
Thanks for the work. 

We're working towards a new, coherent appearance for Jenkins and this has been ongoing for quite a while now. The new icon set introduced into Jenkins last year aims to be simpler, cleaner and more scalable than the previous Tango set and I think it's been adopted very well by core/plugins alike.

I understand the lack of colour in the icons is a downside for some users, one of the goals with the ongoing Jenkins work is to use [colour more sparingly and instead use it for important actions/information (e.g. build colour, warnings, failures)](https://weekly.ci.jenkins.io/design-library/Colors/). 

One of the issues Jenkins currently faces is the slurry of icons on every page, where I can understand that colour can help differentiate them. I personally see applying colour to the iconset as fixing the symptom rather than the cause, I'd like to reduce the amount of icons/menus on each page instead and I think that would really benefit the Jenkins UX. Check the [current prototype](https://jenkins-redesign.vercel.app/project) out if you're interested.

-

I'd love to see this as a new theme for Jenkins (the more the merrier in my opinion), but I don't think replacing what we have in Jenkins is the right approach, and would just introduce inconsistencies at a time where a clear vision is needed.

**Comment by Tampa** [comment]
Oh yeah I saw that prototype before. Shared it with a group of 10+ developers and they are recoiled in horror. Jenkins is a tool, first and foremost, not a design study or phone interface. As far as that goes it's one of the more organized ones in its current form. Comparing to things like GitLab, Nextcloud and visual studio, Jenkins is light on menus and complexity.

I don't understand this push towards this prototype at all, it's a usability nightmare. The icons are barely comprehensible and you have to hover over and weight for and explanation to appear. There is a frankly ridiculous amount of space wasted on padding around buttons and information and yet remains no clearer to what it means. No idea where that "design language" comes from, but it's just awful.

When I first began seeing the changes made over the older interface I immediately got comments from the developers about the changes being disruptive to their workflow given the distinctions were removed from the icons and especially the weather icons. When more changes were made I was basically sat down and asked to revert them or at least change the style sheet enough to make it less confusing to use. See picture I posted.

I get the idea to simplify, but not what you want to simplify given that Jenkins is already one of the lighter tools and not exactly the confusing mess you seem to think it is. What confusions there are does not stem from icons, spacing or anything with the style sheet, it's the illogical arrangement and placement of certain items, features and various pages sporting different layout and design philosophies. You want to declutter and clean up, review the logic on the pages and make sure their structure all follow one central principle not multiple different ones. Concise UI isn't in the design, it's the logic.

An example. I almost never read documentation or tutorials of new software introduced and go in for a first impression completely blind trying to find my bearings. I expect certain things to be in specific places. A top or left menu with a File button for handling files, something for options of preferences. Help section with version info and link to documentation. Software license somewhere in preferences or help. If it has a top ribbon usually there is something there for analytics or preparing data. There are many things that one has come to expect most software follows because over the course of decades that arrangement has been somewhat accepted as being at least concise enough for most to find the things they need. The more software follows that the easier it gets for the enduser to use and the commonality means less work for the developer to figure out where to put stuff. The anti-thesis to that being squarely in the park of visual studio with a lot of confusing and not at all logical arrangement of menu items.

I get that the UI needs work on the basis of making it less cumbersome to develop with as it has become a mess on the backend, but vastly changing the UI step by step throwing in hurdles for users every couple months simply creates frustration. Especially when the changes confuse people more.

Lastly. Our brains are good at picking out things that stick out and follow patterns. Having flat, color-less icons, thus is exactly the opposite of what makes a UI easy to use. Similarly adding tons of white space everywhere instead of shingling tables and making concisely spaced items means no patterns are seen requiring double takes for everything you look at. The more distinct the better things are, that's why the JR tells their train drivers to point, verbalize and act, it associates more things with a specific task and the same principle applies to anything we do. Hence the association with a icon and button text is going to engrave itself a lot better into the brain than a simple flat icon.

If you wish to continue this path I whole-heatedly hope it is going to be in a way that at least provides a way for endusers to change icon sets or still make css adjustments, because I already know I will have my inbox and meeting schedule full with angry developers the moment that prototype goes live in its current form and I will just move away from Jenkins entirely despite the alternative, GitLab, not exactly being the posterchild of readability either.

**Comment by NotMyFault** [comment]
Hey,

Thanks for the PR. We're not interested in reversing the work done (as explained by Jan Faracik above) to get us back into the same state as with the Tango icons.
Therefore, I'm going ahead and closing the PR.

**Comment by Tampa** [comment]
I mean if you want to completely fold into the material flat colorless masses of borderline ambiguous shapes be my guest, but I can speak from experience that developers want consistency and ease of use. They care very little how fancy the UI might be or how great it animates. Turning jenkins into a design study over trying to build usability, well if you want to lose the last users to gitlab and github, sure. I'll certainly be looking at switching should that prototype make prod, because I'd rather axe yet another piece of devops over dealing with upset developers when they can't find a single button anymore. Sorry to be so blunt, but looks are really the last thing anyone cares for, usability, stability and performance are the keys.

**Pull request closed without merging**